polyfactory/factories/base.py

```diff
@@ -642,18 +651,27 @@ def get_field_value(  # noqa: C901, PLR0911, PLR0912
             return cls.get_constrained_field_value(annotation=unwrapped_annotation, field_meta=field_meta)
 
         if BaseFactory.is_factory_type(annotation=unwrapped_annotation):
+            if not field_build_parameters and unwrapped_annotation in build_context["seen_models"]:
+                return None
+
             return cls._get_or_create_factory(model=unwrapped_annotation).build(
+                build_context,
                 **(field_build_parameters if isinstance(field_build_parameters, Mapping) else {}),
             )
 
         if BaseFactory.is_batch_factory_type(annotation=unwrapped_annotation):
             factory = cls._get_or_create_factory(model=field_meta.type_args[0])
             if isinstance(field_build_parameters, Sequence):
-                return [factory.build(**field_parameters) for field_parameters in field_build_parameters]
+                return [factory.build(build_context, **field_parameters) for field_parameters in field_build_parameters]
+
+            if field_meta.type_args[0] in build_context["seen_models"]:
```

Comment:
Favour doing this check after checking for fields parameters as field parameters may provide a way to avoid recusion

Reply:
We need an 'escape hatch' for recursive types. For thing like, union using other value seems sensible. For list, using an empty list feels sensible so another factory create call isn't made. Alternative approach here would be omit the value entirely like in the other case.